include/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -99,7 +104,7 @@ MEMORY
     CCM                   (rw) : ORIGIN = DT_REG_ADDR(DT_CHOSEN(zephyr_ccm)), LENGTH = DT_REG_SIZE(DT_CHOSEN(zephyr_ccm))
 #endif
 #if DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
-    DTCM                  (rw) : ORIGIN = DT_REG_ADDR(DT_CHOSEN(zephyr_dtcm)), LENGTH = DT_REG_SIZE(DT_CHOSEN(zephyr_dtcm))
+    DTCM                  (rw) : ORIGIN = DTCM_ADDR, LENGTH = DTCM_SIZE
```

Comment:
For coherency sake, I wonder if it would makes sense to do the same for ccm, so that ccm, itcm and dtcm are fully aligned following your change.

Reply:
Sure, done. I don't have hardware to test it though, will you please check it?

Reply:
Tested ok on nucleo_f429z (using samples/boards/olimex_stm32_e407/ccm)